src/main/java/com/aws/greengrass/tes/CredentialRequestHandler.java

```diff
@@ -174,6 +179,44 @@ public void handle(final HttpExchange exchange) throws IOException {
         }
     }
 
+    private byte[] getCredentialsWithTimeout(int timeout, TimeUnit timeUnit) throws TimeoutException {
+        TESCache cacheEntry = tesCache.get(iotCredentialsPath);
+        CompletableFuture<Void> future = null;
+        synchronized (cacheEntry) {
+            if (areCredentialsValid(cacheEntry)) {
+                return cacheEntry.credentials;
+            }
+            CompletableFuture<Void> newFut = new CompletableFuture<>();
+            // "take the lock" by immediately setting the future non-null while inside the sync block
+            if (!cacheEntry.future.compareAndSet(null, newFut)) {
+                future = cacheEntry.future.get();
+            }
+        }
+        if (future != null) {
+            LOGGER.atDebug().kv(IOT_CRED_PATH_KEY, iotCredentialsPath)
+                    .log("IAM credentials not found in cache or already expired. A request to fetch new credentials "
+                            + "is already ongoing, waiting for it to complete.");
+            try {
+                // block along with any other threads so we don't send multiple requests
+                if (timeout == 0) {
+                    future.get();
+                } else {
+                    future.get(timeout, timeUnit);
+                }
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+            } catch (ExecutionException ignore) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An exception has been ignored. The catch-block hides information about the stack trace. Not logging or forwarding the stack trace of an exception can mask unexpected errors and complicated debugging.

**Fix**: Use a catch block to handle the exception and write a log message declaring that the exception happened. 
[Learn more.](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)

Suggested remediation:
Handle all exceptions by creating a log for each exception using the message thrown.
```
@@ -209,2 +209,3 @@
                 // We never complete the future exceptionally
+                log.error(&#34;ExecutionException is caught.&#34;, ignore);
             }

```